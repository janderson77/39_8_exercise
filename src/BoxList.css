.BoxList {
	text-align: center;
}

.Boxlist-form-container {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
}

.Boxlist-box-container {
	display: flex;
	align-items: baseline;
	justify-content: flex-start;
	margin: 1rem 2rem;
	width: 100%;
	max-width: 1000px;
	flex-wrap: wrap;
}
