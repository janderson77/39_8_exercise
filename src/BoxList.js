import React, {useState} from 'react'
import BoxForm from './BoxForm'
import Box from './Box'
import {v4 as uuid} from 'uuid'
import './BoxList.css'

const BoxList = () => {
    const INITIAL_STATE = [];
    const [boxes, setBoxes] = useState(INITIAL_STATE)
    const addBox = (newBox) => {
        setBoxes(boxes => [...boxes, {...newBox, id: uuid()}])
    }
    const handleRemove = (e) => {
        const id = e.target.parentNode.id
        const newBoxes = boxes.filter((box) => box.id !== id)
        setBoxes(newBoxes)
    }
    return(
        <div className="Boxlist">
            <div className="Boxlist-form-container">
                <BoxForm addBox={addBox}/>
            </div>
            <div className="Boxlist-box-container">
                {boxes.map(({id, color, width, height}) => <Box id={id} color={color} width={width} height={height} key={id} handleRemove={handleRemove} />)}
            </div>
        </div>
    )
}

export default BoxList