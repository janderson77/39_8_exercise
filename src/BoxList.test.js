import React from "react";
import { render, fireEvent, wait } from "@testing-library/react";
import BoxList from "./BoxList";

function addBox(boxList, height = "200px", width = "200px", color = "peachpuff") {
  const heightInput = boxList.getByLabelText("Height");
  const widthInput = boxList.getByLabelText("Width");
  const Input = boxList.getByLabelText("Color");
  fireEvent.change(Input, { target: { value: color } });
  fireEvent.change(widthInput, { target: { value: width } });
  fireEvent.change(heightInput, { target: { value: height } });
  const button = boxList.getByText("Add Box!");
  fireEvent.click(button);
}

it("renders without crashing", function() {
  render(<BoxList />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("can add a new box", function() {
  const boxList = render(<BoxList />);


  expect(boxList.queryByText("X")).not.toBeInTheDocument();

  addBox(boxList);


  const removeButton = boxList.getByText("X");
  expect(removeButton).toBeInTheDocument();
  expect(removeButton.parentElement).toHaveStyle(`
    background-color: peachpuff;
    width: 200px;
    height: 200px;
  `);

  expect(boxList.getAllByDisplayValue("")).toHaveLength(3);


});

it("can remove a box", function() {
  const boxList = render(<BoxList />);
  addBox(boxList);

  const removeButton = boxList.getByText("X");


  fireEvent.click(removeButton);
  expect(removeButton).not.toBeInTheDocument();
});
