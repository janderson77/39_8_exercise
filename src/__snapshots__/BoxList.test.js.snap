// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Boxlist"
  >
    <div
      class="Boxlist-form-container"
    >
      <form>
        <label
          for="color"
        >
          Color 
        </label>
        <input
          id="color"
          name="color"
          placeholder="Color!"
          type="text"
          value=""
        />
        <label
          for="width"
        >
          Width 
        </label>
        <input
          id="width"
          name="width"
          placeholder="Width!"
          type="text"
          value=""
        />
        <label
          for="height"
        >
          Height 
        </label>
        <input
          id="height"
          name="height"
          placeholder="Height!"
          type="text"
          value=""
        />
        <button>
          Add Box!
        </button>
      </form>
    </div>
    <div
      class="Boxlist-box-container"
    />
  </div>
</DocumentFragment>
`;
